### PR TITLE
Intake

### PR DESCRIPTION
**Pull request by rbhog**

### CODE REVIEW

File: src/main/kotlin/org/usfirst/frc/team4099/robot/Robot.kt [hunk 1]
```diff
@@ -26,7 +33,17 @@ class Robot : IterativeRobot() {
 
     override fun robotInit() {
         try {
+            CameraServer.getInstance().startAutomaticCapture()
```
Comment (atzho) [inline src/main/kotlin/org/usfirst/frc/team4099/robot/Robot.kt]: Is this from last year? Doesn't really belong on this branch

File: src/main/kotlin/org/usfirst/frc/team4099/robot/subsystems/Intake.kt [hunk 2]
```diff
@@ -1,8 +1,103 @@
...
+ * @constructor Creates the Intake subsystem
+ *
+ */
 class Intake private constructor() : Subsystem {
-    fun outputToSmartDashboard()
-    fun stop()
-    fun zeroSensors()
 
-}
+    private val Talon = Talon(Constants.Intake.INTAKE_TALON_ID)
```
Comment (atzho) [inline src/main/kotlin/org/usfirst/frc/team4099/robot/subsystems/Intake.kt]: variable names should start with lower case

File: src/main/kotlin/org/usfirst/frc/team4099/robot/subsystems/Intake.kt [hunk 3]
```diff
@@ -1,8 +1,103 @@
...
+            intakeState = IntakeState.STOP
+        }
+
+        /**
+         * Sets Intake to -1 if pulling in, to 0 if stationary, and 1 if pushing out
+         */
+        override fun onLoop() {
+            synchronized(this@Intake) {
+                //                println("Ribbon switch pressed: $switchPressed")
+                if (intakeState == IntakeState.IN) {
```
Comment (atzho) [inline src/main/kotlin/org/usfirst/frc/team4099/robot/subsystems/Intake.kt]: this if block is unnecessary, remove it

File: src/main/kotlin/org/usfirst/frc/team4099/robot/Robot.kt [hunk 4]
```diff
@@ -88,7 +105,21 @@ class Robot : IterativeRobot() {
...
-
+            if (intake.up && controlboard.lowerIntake) {
+                intake.up = false
+                println("Lowering intake")
+            } else if (!intake.up && controlboard.lowerIntake) {
+                intake.up = true
+                println("Raising intake")
+            }
+
+            intake.intakeState = when {
```
Comment (atzho) [inline src/main/kotlin/org/usfirst/frc/team4099/robot/Robot.kt]: What does this when block do? Setting intake speed should be the same as changing the intake position (like above) and be handled using if blocks. "when" only takes one argument and therefore can't work for multiple controls

**Review by atzho — CHANGES_REQUESTED** [review]
Make sure to check copy pasted code for what it does; not all of it will be relevant to this year's robot. After making changes, make sure this class has no compileKotlin errors (even if other ones do)

File: src/main/kotlin/org/usfirst/frc/team4099/robot/Robot.kt (hunk 1 above)
Comment (Epicness35) [reply src/main/kotlin/org/usfirst/frc/team4099/robot/Robot.kt]: Yes it is from last year.

**Pull request merged**